integration-test/push_test.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+/* eslint-env mocha */
+
+const assert = require('assert')
+const { describe, it, before } = require('mocha')
+const { PromiseHash } = require('../src/common/util')
+
+const { getTestNodeId } = require('../test/util')
+const { MediachainNode: AlephNode } = require('../src/peer/node')
+const { concatNodeClient, concatNodePeerInfo } = require('./util')
+const { generatePublisherId } = require('../src/peer/identity')
+
+const seedObjects = [
+  {id: 'foo:1', foo: 'bar'},
+  {id: 'foo:2', foo: 'baz'}
+]
+
+function seedStatementsToAleph (alephNode: AlephNode): Promise<Array<string>> {
+  return Promise.all(
+    seedObjects.map(obj =>
+      alephNode.ingestSimpleStatement('scratch.test', obj, { refs: [obj.id] })
```

Comment:
one comment here would be towards separating the test flow and the normal flow, i.e. not using `ingest*` functions to seed test data, but this is probably a separate discussion around testing methodology